### PR TITLE
Add release notes for 0.158

### PR DESCRIPTION
**Pull request by erichwang**

### CODE REVIEW

File: presto-docs/src/main/sphinx/release/release-0.158.rst [hunk 1]
```diff
@@ -7,3 +7,17 @@ General Changes
 
 * Fix regression that could cause high CPU and heap usage on coordinator,
   when processing certain types of long running queries.
+* Fix a bug in``EXPLAIN ANALYZE`` where output columns were incorrectly pruned.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.158.rst]: "Fix a bug" is bad phrasing as it's redundant (basically every fix is a bug fix). How about
```
* Fix incorrect pruning of output columns in ``EXPLAIN ANALYZE``.

File: presto-docs/src/main/sphinx/release/release-0.158.rst [hunk 2]
```diff
@@ -7,3 +7,17 @@ General Changes
...
+* Fix incorrect ordering of ``CHAR`` values, where values with trailing spaces
+  where ordered before values with trailing control-characters.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.158.rst]: Typo: where -> were

File: presto-docs/src/main/sphinx/release/release-0.158.rst [hunk 3]
```diff
@@ -7,3 +7,17 @@ General Changes
 
 * Fix regression that could cause high CPU and heap usage on coordinator,
   when processing certain types of long running queries.
+* Fix a bug in``EXPLAIN ANALYZE`` where output columns were incorrectly pruned.
+* Fix incorrect ordering of ``CHAR`` values, where values with trailing spaces
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.158.rst]: This is a run-on sentence

File: presto-docs/src/main/sphinx/release/release-0.158.rst [hunk 4]
```diff
@@ -7,3 +7,17 @@ General Changes
...
+  where ordered before values with trailing control-characters.
+* Fix issue which could cause queries with non-remotely accessible splits to fail.
+* Validate resource groups configuration file when server starts up, to ensure
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.158.rst]: Remove comma
"when server starts up" -> "on startup"

**Comment by electrum** [comment]
Remove the "Resolves:" line from the commit message

**Comment by erichwang** [comment]
Updated with fixes and @martint's notes

File: presto-docs/src/main/sphinx/release/release-0.158.rst [hunk 5]
```diff
@@ -7,3 +7,21 @@ General Changes
...
+* Fix failure of queries that have non-remotely accessible splits.
+* Fix non-linear parsing issue for certain SQL expressions.
+* Fix case-sensitivity issues when operating on columns of ``ROW`` data type.
+* Validate resource groups configuration file on startup. This ensures that all
+  selectors reference a configured resource group.
+* Add experimental on-disk merge sort for aggregations. This can be enabled with
+  the ``experimental.spill-enabled`` configuration flag.
+* Push down predicates for ``DECIMAL``, ``TINYINT``, ``SMALLINT`` and ``REAL`` data types.
+* Produce simpler query plans, which can reduce data transfers over the network and reduce CPU requirements.
+* Return ``NULL`` from ::func::`element_at` for out-of-range indices instead of failing.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.158.rst]: I think this should be
```
:func:`element_at`

File: presto-docs/src/main/sphinx/release/release-0.158.rst [hunk 6]
```diff
@@ -7,3 +7,21 @@ General Changes
...
+  were ordered before values with trailing control-characters.
+* Fix failure of queries that have non-remotely accessible splits.
+* Fix non-linear parsing issue for certain SQL expressions.
+* Fix case-sensitivity issues when operating on columns of ``ROW`` data type.
+* Validate resource groups configuration file on startup. This ensures that all
+  selectors reference a configured resource group.
+* Add experimental on-disk merge sort for aggregations. This can be enabled with
+  the ``experimental.spill-enabled`` configuration flag.
+* Push down predicates for ``DECIMAL``, ``TINYINT``, ``SMALLINT`` and ``REAL`` data types.
+* Produce simpler query plans, which can reduce data transfers over the network and reduce CPU requirements.
```
Comment (electrum) [inline presto-docs/src/main/sphinx/release/release-0.158.rst]: The phrase "produce simpler query plans" doesn't mean anything to the end user. Or worse, they will think it means something completely different.

**Comment by electrum** [comment]
Merged, thanks!

**Comment by erichwang** [comment]
You're welcome

**Pull request closed without merging**